### DISCUSSION TITLE
Add PHP support

### DISCUSSION THREAD

**Discussion by Grosso2020** [GetPublii/Publii#1377] [Ideas]
Hello.

I understand that Publii does not support PHP code and generates pure HTML, but I would really like to have PHP support. I want to implement my own comment system, system of likes and views of posts into the site, but I can’t do this because I need PHP. Perhaps you will consider my request in the future. I think many developers will thank you!

**Comment by internettips** [comment]
There are already at least 4 Publii plugins -- ready to use -- that provide commenting functionality. [https://marketplace.getpublii.com/plugins/](url) Simples :-)

**Comment by dziudek** [comment]
@Grosso2020 - you can use one of the many comments systems dedicated for static websites. Also there is no need to add PHP support in Publii - you can still include PHP files in your theme if you need - but you won't be able to test them during preview - we have no plans to add local PHP server for preview - it is absolutely of our project scope.

**Reply by Grosso2020** [reply]
Thank you. I figured out how to connect PHP and added my view counter. I hope I can add my own comment system. I would not like to use third party scripts

**Reply by dziudek** [reply]
You can always prepare REST API in PHP and use it in HTML files through AJAX

**Reply by candidexmedia** [reply]
> I hope I can add my own comment system. I would not like to use third party scripts

@Grosso2020 There are many open source commenting platforms that work with static sites and use scripts that you can self-host:
- https://waline.js.org/en/
- https://comentario.app/en/
- https://cusdis.com/ (has a Publii plugin)
- There are even solutions out there that use Mastodon discussions as comments
